File: src/apis/shortcut.js
import { invoke } from "@tauri-apps/api/core"

export async function shortcut_to_desktop(file, name, icon, args) {
    return await invoke("shortcut_to_desktop", { file, name, icon, args })
}

export async function shortcut_to_startup(file, name, icon, args) {
    return await invoke("shortcut_to_startup", { file, name, icon, args })
}

File: src-tauri/src/apis/shortcut.rs
use crate::errors::Result;
use services::shortcut;
use utils::shortcut::ShortCutArgs;

#[tauri::command(async)]
pub fn shortcut_to_desktop(
    file: Option<&str>,
    name: Option<&str>,
    icon: Option<&str>,
    args: Option<ShortCutArgs>,
) -> Result<()> {
    Ok(shortcut::shortcut_to_desktop(file, name, icon, args)?)
}

#[tauri::command(async)]
pub fn shortcut_to_startup(
    file: Option<&str>,
    name: Option<&str>,
    icon: Option<&str>,
    args: Option<ShortCutArgs>,
) -> Result<()> {
    Ok(shortcut::shortcut_to_startup(file, name, icon, args)?)
}
